recipe-box-backend/src/main/java/com/endava/recipebox/service/impl/RecipeServiceImpl.java

```diff
@@ -172,4 +172,22 @@ public String updateRecipe(RecipeEditRequestDTO recipeAddRequestDTO, Long userId
 
         return "Recipe update successfully!";
     }
+
+    @Override
+    public String deleteRecipe(Long recipeId, Long userId) {
+        Recipe recipe = getRecipeById(recipeId);
```

Comment:
What happens if the recipe is not found?

Reply:
getRecipeById is this method where I assure to throw an error if the recipe is not found.
`    public Recipe getRecipeById(Long recipeId) {
        Optional<Recipe> recipeOptional = recipeRepository.findById(recipeId);
        if (recipeOptional.isEmpty()) {
            throw new BadRequestException("Recipe with ID " + recipeId + " not found.");
        }
        return recipeOptional.get();
    }`